src/client/ws/client.rs

```diff
@@ -257,164 +255,225 @@ impl<Notif> Drop for Subscription<Notif> {
+				// User dropped the sender side of the channel.
+				None => {
+					log::trace!("[backend]: frontend channel dropped; terminate client");
+					break
+				}
+				// User called `notification` on the front-end
+				Some(FrontToBack::Notification { method, params }) => {
+					log::trace!("[backend]: client prepares to send notification");
+					let _ = sender.send_notification(method, params).await;
+				}
+				// User called `request` on the front-end
+				Some(FrontToBack::StartRequest { method, params, send_back }) => {
+					log::trace!("[backend]: client prepares to send request={:?}", method);
+					match sender.start_request(method, params).await {
+						Ok(id) => {
+							if let Err(send_back) = manager.insert_pending_call(id, send_back) {
+								let _ = send_back.send(Err(Error::DuplicateRequestId));
+							}
+						}
+						Err(err) => {
+							log::warn!("[backend]: client request failed: {:?}", err);
+							let _ = send_back.send(Err(Error::TransportError(Box::new(err))));
+						}
 					}
 				}
-			}
-			// User called `subscribe` on the front-end.
-			Either::Left(Some(FrontToBack::Subscribe { subscribe_method, unsubscribe_method, params, send_back })) => {
-				log::trace!(
-					"[backend]: client prepare to start subscription, subscribe_method={:?} unsubscribe_method:{:?}",
-					subscribe_method,
-					unsubscribe_method
-				);
-				match client.start_subscription(subscribe_method, params).await {
-					Ok(id) => {
-						pending_subscriptions.insert(id, (send_back, unsubscribe_method));
+				// User called `subscribe` on the front-end.
+				Some(FrontToBack::Subscribe { subscribe_method, unsubscribe_method, params, send_back }) => {
+					log::trace!(
+						"[backend]: client prepares to start subscription, subscribe_method={:?} unsubscribe_method:{:?}",
+						subscribe_method,
+						unsubscribe_method
+					);
+					match sender.start_subscription(subscribe_method, params).await {
+						Ok(id) => {
+							if let Err(send_back) = manager.insert_pending_subscription(id, send_back, unsubscribe_method) {
+								let _ = send_back.send(Err(Error::DuplicateRequestId));
+							}
+						}
+						Err(err) => {
+							log::warn!("[backend]: client subscription failed: {:?}", err);
+							let _ = send_back.send(Err(Error::TransportError(Box::new(err))));
+						}
 					}
-					Err(err) => {
-						log::warn!("[backend]: client start subscription failed: {:?}", err);
-						let _ = send_back.send(Err(Error::TransportError(Box::new(err))));
+				}
+				// User dropped a subscription.
+				Some(FrontToBack::SubscriptionClosed(sub_id)) => {
+					log::trace!("Closing in subscription: {:?}", sub_id);
+					// NOTE: The subscription may have been closed earlier if
+					// the channel was full or disconnected.
+					if let Some(request_id) = manager.get_request_id_by_subscription_id(&sub_id) {
+						if let Some((_sink, unsubscribe_method)) = manager.remove_subscription(request_id, sub_id.clone()) {
```

Comment:
NOTE: I missed to send a request to the unsubscribe method before.

I should write a test for this but..... not in this PR :)